lib/block_view.go

```diff
@@ -3276,6 +3400,29 @@ func (bav *UtxoView) _connectTransaction(txn *MsgDeSoTxn, txHash *BlockHash,
 				)
 			}
 		}
+		if txn.TxnMeta.GetTxnType() == TxnTypeUnlockStake {
+			if len(utxoOpsForTxn) == 0 {
+				return nil, 0, 0, 0, errors.New(
+					"ConnectTransaction: TxnTypeUnlockStake must return UtxoOpsForTxn",
+				)
+			}
+			utxoOp := utxoOpsForTxn[len(utxoOpsForTxn)-1]
+			if utxoOp == nil || utxoOp.Type != OperationTypeUnlockStake {
+				return nil, 0, 0, 0, errors.New(
+					"ConnectTransaction: TxnTypeUnlockStake must correspond to OperationTypeUnlockStake",
+				)
+			}
+			totalLockedAmountNanos := uint256.NewInt()
+			for _, prevLockedStakeEntry := range utxoOp.PrevLockedStakeEntries {
+				totalLockedAmountNanos, err = SafeUint256().Add(
+					totalLockedAmountNanos, prevLockedStakeEntry.LockedAmountNanos,
+				)
+				if err != nil {
+					return nil, 0, 0, 0, errors.Wrapf(err, "ConnectTransaction: error computing TotalLockedAmountNanos: ")
+				}
+			}
+			desoLockedDelta = big.NewInt(0).Neg(totalLockedAmountNanos.ToBig())
+		}
```

Comment:
Note to self: Does this sanity check require more than just UnlockStake? Why doesn't staking change the desoLockedDelta? How is this being accounted for.